### DISCUSSION TITLE
Support inline configuration in Mermaid elements

### DISCUSSION THREAD

**Discussion by falkoschindler** [zauberzeug/nicegui#1861] [Ideas / Feature Requests]
As pointed out on [StackOverflow](https://stackoverflow.com/questions/77317635/how-to-use-mermaid-style-settings-in-nicegui), it's currently not possible to add an inline configuration to Mermaid code like this:

```py
ui.mermaid('''
%%{ init : { "theme" : "default", "flowchart" : { "curve" : "linear" }}}%%

flowchart LR
    f1 --> f2
    f2 --> f3
    f2 --> f4
''')
```

Somehow this breaks the rendering and results in a JavaScript error
> Uncaught (in promise) UnknownDiagramError: No diagram type detected matching given configuration for text

I inspected the content which is received by the client and it looks alright, so the transmission seems to work. No idea why it confuses the renderer.

**Comment by CrystalWindSnake** [comment]
@falkoschindler It seems that the directive function of mermaid has been deprecated in version v10.5.0+. [mermaid/config/directives](https://mermaid.js.org/config/directives.html#directives)

**Reply by falkoschindler** [reply]
Oh wow, thanks for pointing into the right direction, @CrystalWindSnake!
It turns out the new configuration syntax already works as expected. 😀

```py
ui.mermaid('''
---
config:
    theme: default
    flowchart:
        curve: linear
---

flowchart LR
    f1 --> f2
    f2 --> f3
    f2 --> f4
''')
```

<img width="206" alt="Screenshot 2023-10-26 at 17 51 15" src="https://github.com/zauberzeug/nicegui/assets/5767091/1062038f-f247-4d86-b879-a05e0733580b">

```py
ui.mermaid('''
---
title: Hello Title
config:
  theme: base
  themeVariables:
    primaryColor: "#00ff00"
---
flowchart
	Hello --> World
''')
```

<img width="99" alt="Screenshot 2023-10-26 at 17 51 43" src="https://github.com/zauberzeug/nicegui/assets/5767091/db5c3b89-708a-40bd-86c3-24d53ecfe92b">
